docs/en/ingest-management/fleet/air-gapped.asciidoc

```diff
@@ -138,20 +146,87 @@ NODE_EXTRA_CA_CERTS="/etc/kibana/certs/ca-cert.pem"
 ----
 
 [discrete]
-[[air-gapped-limitations]]
-== Limitations for {agent} upgrades
-
-The <<upgrade-elastic-agent>> feature in technical preview does not currently support network restricted environments without access to artifacts.elastic.co.
-To upgrade {agent}s:
-
-1. Download the new version from the https://www.elastic.co/downloads/elastic-agent[download page]
-2. Place the tarball in a secure internal network location
-3. Use the <<elastic-agent-upgrade-command,`upgrade`>> command with a custom `--source-uri` option
-
-[discrete]
-=== Example
-
-["source", "sh", subs="attributes"]
+[[host-artifact-registry]]
+== Host your own artifact registry for binary downloads
+
+{agent}s must be able to access the {artifact-registry} to download
+binaries during upgrades. By default {agent}s download artifacts from
+`https://artifacts.elastic.co/downloads/`.
+
+To make binaries available in an air-gapped environment, you can host your own
+custom artifact registry, and then configure {agent}s to download binaries
+from it.
+
+. Create a custom artifact registry in a location accessible to your {agent}s:
+.. Download the latest release artifacts from the public {artifact-registry} at
+`https://artifacts.elastic.co/downloads/`. For example, the
+following cURL commands download all the artifacts that may be needed to upgrade
+{agent}s running on Linux. The exact list depends on which integrations you're
+using.
++
+["source","shell",subs="attributes"]
+----
+curl -O https://artifacts.elastic.co/downloads/apm-server/apm-server-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/apm-server/apm-server-{version}-linux-x86_64.tar.gz.sha512
```

Comment:
@michalpristas Is this what you had in mind? Or would it be enough to tell users they need to download the .asc and .sha512 files for each distribution? WDYT?

Reply:
yes that's it